Rx.NET/Documentation/adr/0001-multi-platform-packaging.md

```diff
@@ -0,0 +1,106 @@
+
+* Plug-in A
+  * Targets .NET Framework 4.0
+  * Has a reference to `System.Reactive.Linq` v3.0
+* Plug-in B 
+  * Targets .NET Framework 4.5
+  * Has a reference to `System.Reactive.Linq` v3.0
+
+Suppose the host application runs in .NET 4.5, and it loads Plug-in A. This is OK because loading of .NET 4.0 components in a .NET 4.5 host process is supported. This in turn will load `System.Reactive.Linq` v3.0. Since Plug-in A was built for .NET 4.0, it will have a copy of the `System.Reactive.Linq.dll` that was in the `lib\net40` folder in the `System.Reactive.Linq` NuGet package. So it gets to use the version it was built for.
+
+So far, so good.
+
+Now suppose the host loads Plug-in B. It will also want to load `System.Reactive.Linq`, but the CLR will notice that it has already loaded that component. The critical detail here is that the `System.Reactive.Linq.dll` files in the `lib\net40` and `lib\net45` folders have **the same name, public key token, and version number**. The .NET Framework assembly loader therefore considers them to have the same identity. So when Plug-in B comes to try to use `System.Reactive.Linq.dll`, the CLR will just let it use the same one that was loaded on behalf of Plug-in A. But if Plug-in B is relying on any functionality specific to the `net45` version of that component, it will be disappointed, with runtime errors such as `MethodNotFoundException`.
+
+This tends not to be an issue with modern versions of .NET because we have [`AssemblyLoadContext`]([https://learn.microsoft.com/en-us/dotnet/core/dependency-loading/understanding-assemblyloadcontext) to solve this very problem. But with .NET Framework, this was a serious problem.
+
+In 2016, Rx attempted to address this in [PR #212](https://github.com/dotnet/reactive/pull/212) by giving a slightly different version number to each target. For example, the `net45` DLL had version 3.0.1000.0, while the `net451` DLL was 3.0.2000.0, and so on. This plan is described in [issue #205](https://github.com/dotnet/reactive/issues/205).
+
+At the time this was proposed, it was [acknowledged](https://github.com/dotnet/reactive/issues/205#issuecomment-228577028) that there was a potential problem with binding redirects. Binding redirects often specify version ranges, which means if you upgrade 3.x to 4.x, it's possible that 3.0.2000.0 would get upgraded to 4.0.1000.0, which could actually mean a downgrade in surface area (because the x.x.2000.0 versions might have target-specific functionality that the x.x.2000.0 versions do not).
+
+These concerns turned out to be valid. [Issue #305](https://github.com/dotnet/reactive/issues/305) shows one example of the baffling kinds of problems that could emerge.
+
+If I've understood correctly, part of the thinking here is that if all of Rx is in a single DLL, you won't have any problems in which different bits of Rx seems to disagree about which version is in use.
+
+### Revisiting the decisions in 2023
```

Comment:
Could you add in brackets (.NET 7.0) so in the future we know what version of .NET was current at the time?